File: Models/ViewModels/CompanyVM.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace GoWheels_WebAPI.Models.ViewModels
{
    public class CompanyVM : BaseModelVM
    {
        [Required]
        public string? Name { get; set; }
        [Required]
        public string? IconImage { get; set; }

        [JsonPropertyOrder(100)]
        public List<CarTypeDetailVM> carTypeDetail { get; set; } = new List<CarTypeDetailVM>();

    }
}

File: FE/src/Model/CompanyVM.js
// src/models/CompanyVM.js

import CarTypeDetailVM from './CarTypeDetailVM';
import BaseModel from './BaseModelVM';
export default class CompanyVM extends BaseModel{
  constructor(data = {}) {
    super(data.id || null, data.createdById, data.createdOn, data.modifiedById, data.modifiedOn, data.isDeleted);
    this.name = data.name || '';
    this.iconImage = data.iconImage || '';
    this.carTypeDetail = (data.carTypeDetail || []).map(detail => new CarTypeDetailVM(detail));
  }
}
